Formula/nginx-unit.rb

```diff
@@ -0,0 +1,267 @@
+
+  depends_on "openssl"
+  depends_on "pcre"
+
+  def install
+    openssl = Formula["openssl"]
+    pcre = Formula["pcre"]
+
+    system "./configure",
+      "--prefix=#{prefix}",
+      "--bindir=#{bin}",
+      "--sbindir=#{bin}",
+      # This tells the unit daemon where it can load modules from. Configuring
+      # the location at runtime is painful. We want this to be in the
+      # $HOMEBREW_PREFIX so that other formulae can link in additional modules. 
+      "--modules=#{HOMEBREW_PREFIX/"lib/unit"}",
+      "--state=#{var/"unit"}",
+      "--log=#{var/"log/unit.log"}",
+      "--pid=#{var/"run/unit.pid"}",
+      "--control=unix:#{var/"run/unit.control.sock"}",
+      "--cc-opt=-I#{pcre.opt_include} -I#{openssl.opt_include}",
+      "--ld-opt=-L#{pcre.opt_lib} -L#{openssl.opt_lib}"
+
+    system "make", "install"
+
+    # Beyond here we want to install modules into the keg, so we'll move them
+    # into the prefix ourselves
+    libunit = lib/"unit"
+    libunit.mkpath
+
+    # There is no system Go
+    #system "./configure", "go"
+    #system "make", "go"
+    #libunit.install "build/go.unit.so"
+
+    system "./configure", "perl"
+    system "make", "perl"
+    libunit.install "build/perl.unit.so"
+
+    # We can't build against the system PHP, it doesn't have the right SAPI
+    #system "./configure", "php"
+    #system "make", "php"
+    #libunit.install "build/php.unit.so"
+
+    system "./configure", "python"
+    system "make", "python"
+    libunit.install "build/python.unit.so"
+
+    system "./configure", "ruby"
+    system "make", "ruby"
+    libunit.install "build/ruby.unit.so"
+  end
+
+  def caveats
+    <<~EOS
+      Once running, you can control unit using the control socket, for example using curl:
+        curl --unix-socket #{var}/run/unit.control.sock http://localhost/
+
+      Modules are available for system perl, python and ruby. The system php doesn't support nginx-unit. You can install additional modules into:
+        #{HOMEBREW_PREFIX/"lib/unit"}
+    EOS
+  end
+
+  plist_options :manual => "unitd"
+
+  def plist; <<~EOS
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+      <dict>
+        <key>Label</key>
+        <string>#{plist_name}</string>
+        <key>RunAtLoad</key>
+        <true/>
+        <key>KeepAlive</key>
+        <false/>
+        <key>ProgramArguments</key>
+        <array>
+            <string>#{opt_bin}/unitd</string>
+            <string>--no-daemon</string>
+        </array>
+        <key>WorkingDirectory</key>
+        <string>#{HOMEBREW_PREFIX}</string>
+      </dict>
+    </plist>
+    EOS
+  end
+
+  test do
+    system "#{bin}/unitd", "--version"
+
+    require "json"
+    require "net/http"
+    require "socket"
+    require "timeout"
+
+    logfile = var/"log/unit.log"
+    logfile.unlink if logfile.exist?
```

Comment:
I think this is permitted by the sandbox, _think_, but it'd be nicer to keep all the filesystem operations inside the temporary directory if possible.

Reply:
It does seem to work (in the sandbox, during my tests) but was mostly because my log file kept growing when re-running the tests. I'll remove it.